r/R/compute.R

```diff
@@ -286,3 +286,8 @@ cast_options <- function(safe = TRUE, ...) {
   )
   modifyList(opts, list(...))
 }
+
+strptime_arrow <- function(..., format, unit){
```

Comment:
I'm not sure how useful this function is since it is a thin wrapper around `call_function()` and we can't set it as an S3 method.. More useful would be to add a version of this in the `nse_funcs` in dplyr-functions.R.

In either case, we should match the `base::strptime()` signature: `function (x, format, tz = "")` with the possible addition of `unit` if that's an Arrow feature. 

Also, should `format` and `unit` have default arguments?

Reply:
Got it.
Arrow function uses `format` and `unit` as an FunctionOption if I understand correctly, haven't found `tz` yet.

I think they should have defaults, yes: `format = "%Y-%m-%d %H:%M:%S"` and `unit = TimeUnit$MICRO/2L/"us"`.

Reply:
@nealrichardson I have trouble with calling `strptime()` function from `nse_funcs` - possible name collision with base. Am I missing something? Thank you!

As for defaults I correct myself, `format` shouldn't have default argument - to match `base::strptime()` signature.

Reply:
What I was suggesting was

```r
nse_funcs$strptime <- function(x, format = "%Y-%m-%d %H:%M:%S", tz = "", unit = "ms") {

}
```

following the model of the other functions there that build Expressions. And if `tz` is not supported somehow, we `stop()` if `tz` is provided.